diff --git a/sys/kern/kern_proc.c b/sys/kern/kern_proc.c
index 8ceca9fd..035140ab 100644
--- a/sys/kern/kern_proc.c
+++ b/sys/kern/kern_proc.c
@@ -2097,7 +2097,7 @@ sysctl_kern_proc_pathname(SYSCTL_HANDLER_ARGS)
 			return (ESRCH);
 	}
 	lwkt_gettoken_shared(&p->p_token);	/* deal with exit race */
-	if (p->p_textnch.ncp) {
+	if (p->p_fd != NULL && p->p_textnch.ncp) {
 		cache_copy(&p->p_textnch, &nch);
 		error = cache_fullpath(p, &nch, NULL, &retbuf, &freebuf, 0);
 		cache_drop(&nch);
